File: src/LAB2/f.py
class Node:
    def __init__(self, val):
        self.val = val
        self.next = None

def insert_node(head, data, position):
    new_node = Node(data)
    if position == 0:
        new_node.next = head
        return new_node

    cur = head
    index = 0
    while cur and index < position - 1:
        cur = cur.next
        index += 1

    new_node.next = cur.next
    cur.next = new_node
    return head

n = int(input())
head = None
tail = None
for _ in range(n):
    val = int(input())
    node = Node(val)
    if not head:
        head = node
        tail = node
    else:
        tail.next = node
        tail = node

data = int(input())
position = int(input())
head = insert_node(head, data, position)

cur = head
out = []
while cur:
    out.append(str(cur.val))
    cur = cur.next
print(' '.join(out))

File: src/LAB2/f.cpp
#include <iostream>
using namespace std;

template<typename T>
class Node {
public:
    T data;
    Node<T> *next;

    Node(T data) {
        this->data = data;
        this->next = NULL;
    }
};

int main() {
    int n;
    cin >> n;

    Node<int> *head = NULL;
    Node<int> *tail = NULL;

    for (int i = 0; i < n; i++) {
        int value;
        cin >> value;
        Node<int> *currentNode = new Node<int>(value);

        if (head == NULL) {
            head = currentNode;
            tail = currentNode;
        } else {
            tail->next = currentNode;
            tail = currentNode;
        }
    }

    int pos, value;
    cin >> value;
    cin >> pos;

    Node<int> *newNode = new Node<int>(value);

    if (pos == 0) {
        newNode->next = head;
        head = newNode;
    } else {
        Node<int> *currentNode = head;
        for (int i = 0; i< pos - 1 && currentNode != NULL; i++) {
            currentNode = currentNode->next;
        }
        if (currentNode != NULL) {
            newNode->next = currentNode->next;
            currentNode->next = newNode;
        }
    }

    Node<int> *printNode = head;
    while (printNode != NULL) {
        cout << printNode->data << " ";
        printNode = printNode->next;
    }
}